Dockerfile

```diff
@@ -85,9 +92,37 @@ RUN mkdir -p /etc/apt/keyrings && \
     echo "deb [arch=`dpkg --print-architecture` signed-by=/etc/apt/keyrings/microsoft.gpg] https://packages.microsoft.com/repos/azure-cli/ $AZ_DIST main" | \
     tee /etc/apt/sources.list.d/azure-cli.list && \
     apt-get update && \
-    # Install a specific version
     apt-get install --no-install-recommends -y azure-cli=$AZURECLI_VERSION-1~$AZ_DIST
 
+# PowerShell Installation
+ARG PS_VERSION=7.4.4
+ARG PS_PACKAGE=powershell_${PS_VERSION}-1.deb_amd64.deb
+ARG PS_PACKAGE_URL=https://github.com/PowerShell/PowerShell/releases/download/v${PS_VERSION}/${PS_PACKAGE}
+ARG PS_INSTALL_VERSION=7
+
+RUN curl -sSL ${PS_PACKAGE_URL} -o /tmp/powershell.deb && \
+    apt-get install --no-install-recommends -y /tmp/powershell.deb && \
+    rm /tmp/powershell.deb
+
+ENV DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=false \
+    LC_ALL=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    PS_INSTALL_FOLDER=/opt/microsoft/powershell/$PS_INSTALL_VERSION \
+    PSModuleAnalysisCachePath=/var/cache/microsoft/powershell/PSModuleAnalysisCache/ModuleAnalysisCache \
+    POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-Ubuntu-22.04
+
+RUN locale-gen $LANG && update-locale && \
+    export POWERSHELL_TELEMETRY_OPTOUT=1 && \
+    chmod a+x,o-w ${PS_INSTALL_FOLDER}/pwsh && \
+    ln -sf ${PS_INSTALL_FOLDER}/pwsh /usr/bin/pwsh && \
+    pwsh -NoLogo -NoProfile -Command " \
+        \$ErrorActionPreference = 'Stop' ; \
+        \$ProgressPreference = 'SilentlyContinue' ; \
+        while(!(Test-Path -Path \$env:PSModuleAnalysisCachePath)) {  \
```

Comment:
Will this leads to an infinitive loop?
I suggest adding a `maxTries` for this while loop. (e.g: maxTries=20) then we will fail the build if PSModuleAnalysisCachePath is still not available.)

Reply:
Thank you Tung, for your valuable feedback. I have conditioned the while loop with maxTries. 
Python is not my skill but I have adjusted the code in check_latest_version, update_latest_version and tookit_info. Please review it again.

Reply:
Thanks for the quick action, it looks good to me now.